packages/bundler-plugin-core/src/types.ts

```diff
@@ -124,8 +124,17 @@ export interface Options {
      *
      * Defaults to making all sources relative to `process.cwd()` while building.
      */
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    rewriteSources?: (source: string, map: any) => string;
+    rewriteSources?: RewriteSourcesHook;
+
+    /**
+     * Hook to customize source map file resolution.
+     *
+     * The hook is called with the absolute path of the build artifact and the value of the `//# sourceMappingURL=`
+     * comment, if present. The hook should then return an absolute path indicating where to find the artifact's
+     * corresponding `.map` file. If no path is returned or the returned path doesn't exist, the standard source map
+     * resolution process will be used.
+     */
+    resolveSourceMap?: ResolveSourceMapHook;
```

Comment:
Let's not forget to add this option to https://github.com/getsentry/sentry-javascript-bundler-plugins/blob/24191277f07f86488e92e24353305ee09d6a8c71/packages/dev-utils/src/generate-documentation-table.ts#L11 with a good description. Ideally, we also add a short explainer paragraph to the JS doc and the docs in what situations this is beneficial.

Reply:
Thanks for pointing that out, I wasn't aware of that file. It might be good to add a note about it to the `CONTRIBUTING.md` file.